clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -646,7 +646,7 @@ private void handleShareAcknowledgeSuccess(Node fetchTarget,
                                     metadata.topicNames().get(shareAcknowledgeTopicResponse.topicId()));
 
                             acknowledgeRequestState.handleAcknowledgeErrorCode(tip, response.error());
-                            metricsManager.recordLatency(resp.requestLatencyMs());
+                            metricsManager.recordLatency(resp.destination(), resp.requestLatencyMs());
```

Comment:
Pardon me, why we record latency in the loop?

Reply:
I missed it as well, yeah it should be recorded just once for the responce.

Reply:
That's a very good point. Thanks.